src/sync.rs

```diff
@@ -0,0 +1,103 @@
+use config::Config;
+use ignore::Ignore;
+use std::path::PathBuf;
+use std::process::Command;
+
+// TODO add internal version of sync functions with closures as parameters to unit test properly.
+pub fn sync_local_to_remote(local_project_dir_name: &str, config: &Config, ignore: &Ignore) -> Result<(), String> {
```

Comment:
`sync_remote_to_local` and `sync_local_to_remote` seem awfully similar. Most likely they can be unified somehow. At least the `result` part is the same.